connectivity/check/logging.go

```diff
@@ -281,41 +285,58 @@ func (t *Test) Infof(format string, a ...interface{}) {
 	t.logf(info+" "+format, a...)
 }
 
+func (t *Test) failCommon() {
+	t.failed = true
+	t.flush()
+	if t.ctx.params.PauseOnFail {
+		t.log("Pausing after action failure, press the Enter key to continue:")
+		cont := make(chan struct{})
+		go func() {
+			fmt.Scanln()
+			close(cont)
+		}()
+		ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
```

Comment:
Incidentally I think this should ideally check against some broader CLI-wide context that already subscribes to this signal instead of establishing a new signal context here, but the most obvious way I could think to do this is to force all `Fail()` and variations to pass the `ctx` down which is not ideal from a developer perspective.

Reply:
We _could_ store a context in an object somewhere or have a channel that's closed to signal that the test and/or its parents are being terminated.. not sure the best approach.

Reply:
Storing contexts is generally frowned upon as I understand it.

Reply:
> the most obvious way I could think to do this is to force all Fail() and variations to pass the ctx down

yeah i think this is the way, maybe we can just open a separate good first issue 🥰

Reply:
Filed https://github.com/cilium/cilium-cli/issues/2042